### DISCUSSION TITLE
Help creating function with parameters

### DISCUSSION THREAD

**Discussion by erronfr** [supabase/supabase#18486] [Questions] [answered]
Hi, let me start by saying that my sql skills are basic so keen to get this working using the supabase UI.

i have a table that lists timesheets with dates and hours for users.

Im trying to create a function that adds up the total hours or a date and user then write the total hours to another table. Im stuck at trying to setup the Trigger and Functions in Supabase however.

I can create a function okay, which needs to have a Return Type of 'Trigger' otherwise i cant call it from a Row Created Trigger. It tells me i need to use plpgsql in order to use the Trigger Return Type, no problems, but then when i try to save with any Arguments, it says Trigger functions cannot use Arguments.

So how do i pass my user details to the function when a new row is created so the function knows which user to get the total hours for?

![image](https://github.com/supabase/supabase/assets/149036267/b70b1133-71bd-43af-987d-240ac7a6734f)

**Comment by GaryAustin1** [comment, marked as answer]
You can't pass arguments to trigger functions.  This is a Postgres thing not a Supabase thing.
If your user id is the auth UUID then you can just use auth.uid() in your function for the current user.

Edit:  This assumes the table the trigger is on does not have the info in the row as a column.  All columns are passed to the function in an object call NEW.  So in a trigger function you can use NEW.col1 to get the data from that column.

**Reply by erronfr** [reply]
> Edit: This assumes the table the trigger is on does not have the info in the row as a column. All columns are passed to the function in an object call NEW. So in a trigger function you can use NEW.col1 to get the data from that column

This sounds like what i need, however i cant find any documentation or find anything on google for this, do you have any links that may help here?

**Reply by GaryAustin1** [reply]
On Postgres trigger functions?  Lots of info out there as it is not supabase specific.  Just google Postgres trigger functions.
Supabase has an example https://supabase.com/docs/guides/database/postgres/triggers .

**Reply by erronfr** [reply]
Much appreciated, got it working now :)
